packages/cli/internal/pkg/cli/workflow/workflow_run_test.go

```diff
@@ -160,17 +167,33 @@ func (s *WorkflowRunTestSuite) BeforeTest(_, _ string) {
 					},
 				},
 			},
+			testContext2Name: {
+				Engines: []spec.Engine{
+					{
+						Type:   "nextflow",
+						Engine: "nextflow",
+					},
+				},
+			},
 		},
 	}
 
-	s.wfInstance = ddb.WorkflowInstance{
+	s.wfInstance1 = ddb.WorkflowInstance{
 		RunId:        testRun1Id,
 		WorkflowName: testLocalWorkflowName,
 		ContextName:  testContext1Name,
 		ProjectName:  testProjectName,
 		UserId:       testUserId,
 	}
 
+	s.wfInstance2 = ddb.WorkflowInstance{
```

Comment:
Why do we need another WorkflowInstance?

Reply:
We need a different context to test engineOptions since it can only be used with head node engines.